src/server/graphql/models/Team/archiveTeam/archiveTeam.js

```diff
@@ -0,0 +1,58 @@
+import getRethink from 'server/database/rethinkDriver';
+import {
+  requireSUOrTeamMember,
+  requireWebsocket
+} from 'server/utils/authorization';
+import {handleSchemaErrors} from 'server/utils/utils';
+import {
+  GraphQLNonNull,
+  GraphQLBoolean,
+} from 'graphql';
+import {TeamInput} from '../teamSchema';
+import shortid from 'shortid';
+import {TEAM_ARCHIVED} from 'universal/utils/constants';
+import archiveTeamValidation from './archiveTeamValidation';
+
+export default {
+  type: GraphQLBoolean,
+  args: {
+    updatedTeam: {
+      type: new GraphQLNonNull(TeamInput),
+      description: 'The input object containing the teamId and any modified fields'
+    }
+  },
+  async resolve(source, {updatedTeam}, {authToken, socket}) {
+    const r = getRethink();
+
+    // AUTH
+    requireSUOrTeamMember(authToken, updatedTeam.id);
+    requireWebsocket(socket);
+
+    // VALIDATION
+    const {errors, data: {id, name, isArchived}} = archiveTeamValidation()(updatedTeam);
+    handleSchemaErrors(errors);
+
+    // RESOLUTION
+    await r.table('Team').get(id).update({isArchived});
+
+    setTimeout(async () => {
```

Comment:
You've got the logic correct here, but I would prefer these database interactions to be changed in the following way:

1. It be made a single query, rather than 3
2. It not be placed in a `setTimeout` handler (it'll be easier to test)...
3. ...and speaking of testing, that you add a unit test for this function

If any of these things are unclear/challenging please ask questions. We're happy to help here!